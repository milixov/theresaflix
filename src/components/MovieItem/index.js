import React from 'react';
import {StyleSheet, View, Text, Image, TouchableOpacity} from 'react-native';

import {IMAGE_BASE_URL} from '../../config/enums';
const IMAGE_RESOLUTION = `${IMAGE_BASE_URL}/w500`;

const MovieItem = ({title, vote_average, poster_path, onPress}) => {
    return (
        <TouchableOpacity onPress={onPress}>
            <View style={styles.container}>
                <View style={styles.textWrapper}>
                    <Text style={styles.title}>{title}</Text>
                    <Text style={styles.rate}>{vote_average}</Text>
                </View>
                <Image
                    style={styles.image}
                    source={{uri: IMAGE_RESOLUTION + poster_path}}
                />
            </View>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    container: {
        marginHorizontal: 16,
        borderRadius: 8,
        backgroundColor: '#FFFFFF',
        overflow: 'hidden',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    textWrapper: {
        padding: 24,
        flexShrink: 0,
    },
    title: {
        fontSize: 18,
        maxWidth: 200,
    },
    rate: {
        marginTop: 8,
        paddingVertical: 4,
        paddingHorizontal: 8,
        backgroundColor: 'orange',
        borderRadius: 4,
        alignSelf: 'flex-start',
    },
    image: {
        width: 100,
        height: '100%',
        maxWidth: 100,
        flexGrow: 1,
    },
});

export default MovieItem;
