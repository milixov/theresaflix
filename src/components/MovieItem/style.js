import THEME from '../../style';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
    container: {
        marginHorizontal: 16,
        borderRadius: 8,
        backgroundColor: THEME.COLOR.WHITE,
        overflow: 'hidden',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    textWrapper: {
        padding: THEME.DIMEN.SCREEN_PADDING,
        flexShrink: 0,
    },
    title: {
        fontSize: 18,
        maxWidth: 200,
    },
    rate: {
        marginTop: 8,
        paddingVertical: 4,
        paddingHorizontal: 8,
        backgroundColor: THEME.COLOR.ORANGE,
        borderRadius: 4,
        alignSelf: 'flex-start',
    },
    image: {
        width: 100,
        height: '100%',
        maxWidth: 100,
        flexGrow: 1,
    },
});
