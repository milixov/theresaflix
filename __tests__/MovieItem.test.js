import React from 'react';
import MovieItem from '../src/components/MovieItem';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<MovieItem />).toJSON();
    expect(tree).toMatchSnapshot();
});
