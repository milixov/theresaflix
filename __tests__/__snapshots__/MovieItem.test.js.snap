// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 8,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 16,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flexShrink": 0,
          "padding": 24,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "maxWidth": 200,
          }
        }
      />
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "orange",
            "borderRadius": 4,
            "marginTop": 8,
            "paddingHorizontal": 8,
            "paddingVertical": 4,
          }
        }
      />
    </View>
    <Image
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w500undefined",
        }
      }
      style={
        Object {
          "flexGrow": 1,
          "height": "100%",
          "maxWidth": 100,
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
